### ISSUE TITLE
TextInput Lag, Android 13

### ISSUE THREAD

**Issue by DenisPostnikov** [facebook/react-native#35735]
### Description

Device Pixel, Android 13

Case 1:
During typing a of text observed lagging.

Case 2:
After typing a lot of text into a Text Input, we start freezing.

Adding **autoCorrect={false}**,  **keyboardType="visible-password"** and **returnKeyType='none'** not helped.

### Version

0.68.5

### Output of `npx react-native info`

-

### Steps to reproduce

Start typing in TextInput on the Pixel device.

### Snack, code example, screenshot, or link to a repository

![repr](https://user-images.githubusercontent.com/18422856/209835540-0a068992-88e0-47f9-8f75-92819c543347.gif)

**Comment by stelmakhivan** [comment]
I've faced the same issue with Android 13, Pixel 6
Also `autoCorrect={false}`,` keyboardType="visible-password"` and `returnKeyType='none'` not helped.

**Comment by Vijayakumarktm** [comment]
Any update in this issue? 
I'm also facing issue in Android 13.

**Comment by Kishore0297** [comment]
me too facing issue in android 13

**Comment by cortinico** [comment]
Potentially related to #35590
What happens if you copy-paste a wall of text?

**Comment by Vijayakumarktm** [comment]
@cortinico No solution in #35590 also.
Any other solution to fix it?

**Comment by welch174** [comment]
Try drawing your list upside down, and then the contents in the list upside down. https://github.com/facebook/react-native/issues/35350

**Comment by stelmakhivan** [comment]
I used inverted `FlatList` with `TextInput` on one screen and had this issue...
My current `react-native` version is 0.66
I've tried `<TextInput autoCorrect={Platform.OS === 'ios' || Platform.Version <= 32} spellCheck={Platform.OS === 'ios' || Platform.Version <= 32} />`
And this [patch](https://www.npmjs.com/package/patch-package)
```
--- a/node_modules/react-native/Libraries/Lists/VirtualizedList.js
+++ b/node_modules/react-native/Libraries/Lists/VirtualizedList.js
@@ -9,6 +9,7 @@
  */
 
 const Batchinator = require('../Interaction/Batchinator');
+const Platform = require('../Utilities/Platform');
 const FillRateHelper = require('./FillRateHelper');
 const ReactNative = require('../Renderer/shims/ReactNative');
 const RefreshControl = require('../Components/RefreshControl/RefreshControl');
@@ -2118,9 +2119,13 @@ function describeNestedLists(childList: {
 }
 
 const styles = StyleSheet.create({
-  verticallyInverted: {
-    transform: [{scaleY: -1}],
-  },
+  verticallyInverted: Platform.OS === 'android' ?
+    {
+      scaleY: -1,
+    } :
+    {
+      transform: [{scaleY: -1}],
+    },
   horizontallyInverted: {
     transform: [{scaleX: -1}],
   },
```

Seems that those changes fix the issue, but I'm not sure right now. Need more testing on real devices.

**Comment by DenisPostnikov** [comment]
> Try drawing your list upside down, and then the contents in the list upside down. #35350

I used TextInput without Flatlist

**Comment by welch174** [comment]
> > Try drawing your list upside down, and then the contents in the list upside down. #35350
> 
> I used TextInput without Flatlist

Is the view above your textinput using a flatlist or sectionlist?

**Comment by Vijayakumarktm** [comment]
@stelmakhivan Myself using inverted Flatlist and TextInput. Both are in different component. How inverted list is related to this TextInput lag issue?

**Comment by DenisPostnikov** [comment]
> > > Try drawing your list upside down, and then the contents in the list upside down. #35350
> > 
> > 
> > I used TextInput without Flatlist
> 
> Is the view above your textinput using a flatlist or sectionlist?

Nothing from this only View

**Comment by charlesionlake** [comment]
Did anyone have any success fixing this? I'm using react-native-gifted-chat so not so straightforward to invert FlatList of messages causing the issue.

**Comment by JuanAlejandro** [comment]
> > > > Try drawing your list upside down, and then the contents in the list upside down. #35350
> > > 
> > > 
> > > I used TextInput without Flatlist
> > 
> > 
> > Is the view above your textinput using a flatlist or sectionlist?
> 
> Nothing from this only View

Can you share the code you are using to render that list?

**Comment by stelmakhivan** [comment]
I've tried patch-package
https://github.com/facebook/react-native/commit/be69c8b5a77ae60cced1b2af64e48b90d9955be5
Seems the issue is already fixed 4 days ago.

**Comment by DenisPostnikov** [comment]
> > > > > Try drawing your list upside down, and then the contents in the list upside down. #35350
> > > > 
> > > > 
> > > > I used TextInput without Flatlist
> > > 
> > > 
> > > Is the view above your textinput using a flatlist or sectionlist?
> > 
> > 
> > Nothing from this only View
> 
> Can you share the code you are using to render that list?

`<TextInput
    multiline
    editable={!isDisabled}
    placeholder={_(
      isDisabled
        ? 'CHAT_INPUT_PLACEHOLDER_DISABLED'
        : 'CHAT_INPUT_PLACEHOLDER',
    )}
    placeholderTextColor={AppStyles.placeholderColor}
    style={[
      styles.input,
      {maxHeight: conversationChat ? 105 : 140}
    ]}
    value={message}
    onChangeText={this.onMessageChanged}
    onFocus={this.props.onFocusChatInput}
    onBlur={this.props.onBlurChatInput}
  />`

**Comment by ejain** [comment]
> Potentially related to #35590

I can reproduce this issue on a Pixel device that has never seen Grammarly and is running an app using React Native v0.70.9, where that issue has been fixed.

> What happens if you copy-paste a wall of text?

It's not the amount of text (copy-pasting a large block of text is fine), but the number of edits (including deletions). Simply typing and deleting a single character 50x is enough to make the input lag severely.

**Comment by ejain** [comment]
Duplicate of https://github.com/facebook/react-native/issues/35350?

**Comment by holmesjr** [comment]
This is a five-year-old issue: #20119. Do not hold your breath for a solution.

**Comment by holmesjr** [comment]
> This issue is stale because it has been open 180 days with no activity. Remove stale label or comment or this will be closed in 7 days.

So the way bugs in RN are handled is to ignore them and hope nobody comments on them?

**Comment by holmesjr** [comment]
> This issue is stale because it has been open 180 days with no activity. Remove stale label or comment or this will be closed in 7 days.

Not today bot. This issue stays open as long as I'm alive, despite it being totally ignored for years.

**Comment by Ashishpal438** [comment]
i had this same issue for number input in my Credit Card screen. 
For me it worked with  both :  `keyboardType="number-pad"`  or `keyboardType="phone-pad"`
you can also try `inputMode="numeric"`
Found it in this thread : https://github.com/expo/expo/issues/34156

**Comment by holmesjr** [comment]
> This issue is stale because it has been open 180 days with no activity. Remove stale label or comment or this will be closed in 7 days.

Nope, I'm still alive:

https://github.com/facebook/react-native/issues/35735#issuecomment-2323837145

**Comment by holmesjr** [comment]
No, I'm still not unalived. Here's to another ten years of inaction on this.
